### ISSUE TITLE
Error in __init__.py def pytorch_wrap(fn):

### ISSUE THREAD

**Issue by LinChen-Cherry** [facebookresearch/visdom#366]
File "/home/lin/anaconda3/lib/python3.6/site-packages/visdom/__init__.py", line 231, in <genexpr>
    args = (a.cpu().numpy() if type(a).__module__.startswith('torch') else a for a in args)
RuntimeError: Can't call numpy() on Variable that requires grad. Use var.detach().numpy() instead.

I'm not sure, if I can call viz.images() on the Tensor that requires grad. However, when I fix it according to the Err message, it works.

**Comment by lvdmaaten** [comment]
We can simply add the check, right? Something like:
```
processed_args = []
for val in args.values():
   if torch.is_tensor(val):
      val = val.detach() if val.requires_grad else val
      val = val.cpu().numpy()
   processed_args.append(val)
```

**Comment by LinChen-Cherry** [comment]
@JackUrb 
As the docs of pytorch
detach()
    Returns a new Tensor, detached from the current graph.
    The result will never require gradient.

I think the solution preserves behavior for tensors that don't require grad. I test it, it's fine for my project.

**Comment by JackUrb** [comment]
@lvdmaaten is there any reason we would want to avoid detaching all tensors, maybe perf concerns?

**Comment by lvdmaaten** [comment]
As long as the detaching does not happen in-place, I see no harm in always
detaching. Note that detaching is unnecessary when input tensors are
already detached.

**Comment by JackUrb** [comment]
Detaching creates a new view onto the data present in the old tensor. We don't modify this data, so detaching always should be fine. Also I think detaching something that has no associated graph is still a valid operation. 

@LinChen-Cherry would you like to submit a PR?

**Comment by LinChen-Cherry** [comment]
Sure! Let me have a try.

**Comment by JackUrb** [comment]
Fixed in #370. Thanks again @LinChen-Cherry
